rfc/rfc022-discovery-service.md

```diff
@@ -48,16 +48,21 @@ All presentations MUST conform to the Presentation Definition associated with th
 
 Presentations MUST be encoded in JWT format as string ([JWT]).
 
+The protocol identifies the server and client roles. A client can be a server and vice versa. The individual operations define what to do in certain cases.
+
 ### 3.1 Registration
 
-Clients can publish a Verifiable Presentation to the list by sending it in an HTTP POST to the service endpoint.
+Clients can publish a Verifiable Presentation to the server by sending it in an HTTP POST to the service endpoint.
 The HTTP request body MUST be a Verifiable Presentation (in JWT format). The content type of the request MUST be `application/json`.
 
 The server MUST validate the Verifiable Presentation as specified in section 4. If the validation fails, it MUST return a 400 Bad Request response.
-If the validation succeeds, the Verifiable Presentation MUST be added to the list and the server MUST return a 201 Created response.
+If the validation succeeds, the Verifiable Presentation MUST be added to the internal list and the server MUST return a 201 Created response.
+If the Verifiable Presentation is already on the list, the server MUST return a 304 Not Modified response.
+If the server is configured as client, it MUST send the presentation to the server it has configured.
 
 A credential subject (identified by `credentialSubject.id`) MUST NOT appear more than once on the list,
 so a new registration MUST replace the previous one from the same credential subject.
+If two presentations have the same issuance date, the server MUST keep the one with the highest byte value of the sha256 of the presentation. 
```

Comment:
I'd not use issuance date, since they can be out of order with the lamport timestamp. Furthermore, it's not required: if a lamport timestamp is used, it's impossible to have multiple VPs at the same time. Since each lamport timestamp increment = 1 event (aka VP).

With clock timestamps (e.g. unix time), duplicity would be a problem and you would need such a rule, but with lamport timestamps you don't (last write wins).

Reply:
This is the registration part, the server will assign a LC to a submitted VP. A client can submit two VPs out of order/at the same time. The server will have to assign a LC value to both. Given the fact that they should come from the same party, using `nbf` should not lead to any problems.

Reply:
Added note that server must issue timestamps and last timestamp wins.